include/libcyphal/transport/can/delegate.hpp

```diff
@@ -104,6 +105,45 @@ class TransportDelegate
 
     };  // CanardMemory
 
+    /// @brief Utility type with various helpers related to Canard AVL trees.
+    ///
+    /// @tparam Node Type of a concrete AVL tree node.
+    ///
+    template <typename Node>
+    struct CanardConcreteTree
+    {
+        /// @brief Recursively visits each node of the AVL tree, counting total number of nodes visited.
+        ///
+        /// Recursion goes first to the left child, then to the current node, and finally to the right child.
+        /// B/c AVL tree is balanced, the total complexity is `O(n)` and call stack depth
+        /// should not be deeper than `O(log(N))` (max 13 in case of 8192 active message subscriptions),
```

Comment:
Nit but the maximum depth of an AVL tree is $\lt \lceil{} 1.44 \log_2(N+2) - 0.328\rceil{}$, which yields <19 call stack levels

Reply:
thanks, good to know - I will correct it.